src/components/common/CompoundLayout/doesComponentExistIn.ts

```diff
@@ -0,0 +1,20 @@
+import { Children, ReactElement, ReactNode, isValidElement } from "react"
+
+export function doesComponentExistIn(
```

Comment:
I think we don't use this function.

Reply:
I had an ongoing idea about a failsafe to check if the `Layout` is constructed properly. This function would help create this feature. Do you think we need this failsafe? @BrickheadJohnny

Reply:
Yeah, I like the idea, but since this is one of the main components on each page, I think we'd notice any issues with it during development, so we don't need to add this extra logic to it, let's keep it simple.